### DISCUSSION TITLE
Randomness in micrstrucutre

### DISCUSSION THREAD

**Discussion by jackson911007** [kip-hart/MicroStructPy#94] [General]
Hello, 

I have a quick question regarding to the generated mesh structure. I used the bellow code to generate a 2D structure and it works. However, the data (microstructure) I generate for each time is always the same. Is there a way to generate same distribution but different microstructure each time (That means including randomness)?

Thanks, 

Dung-Yi

[Code]

    # Create Materials
    material = {
        'name': 'Matrix',
        'material_type': 'matrix',
        'fraction': 1,
        'shape': 'Circle',
        'size': 
        scipy.stats.lognorm(s=shape, loc=loc, scale=scale)
    }

    # Create Domain
    domain = msp.geometry.Square(side_length=length_of_side, corner=(0, 0))

    # Create List of Un-Positioned Seeds
    seed_area = domain.area
    rng_seeds = {'size': 0}
    seeds = msp.seeding.SeedList.from_info(material,
                                           seed_area,
                                           rng_seeds)

**Comment by kip-hart** [comment]
Hi @jackson911007 do you vary the integer in rng_seeds['size'] and it still outputs the same microstructure?

**Comment by jackson911007** [comment]
Hi Kip-hart. I think I got it by changing the rng_seeds. Thanks for your
help

Best

**Comment by kip-hart** [comment]
Another thing to try is adding an rng seed to the position method. https://docs.microstructpy.org/en/latest/api/seeding/seedlist.html#microstructpy.seeding.SeedList.position

**Comment by jackson911007** [comment]
Thanks for the tip!
